src/parser/core/modules/Gauge/CounterGauge.ts

```diff
@@ -66,25 +96,81 @@ export class CounterGauge extends AbstractGauge {
 	reset() {
 		// NOTE: This assumes counters always reset to their minimum value.
 		// Should that not be the case, probbaly needs a `resetTo` value.
-		this.set(this.minimum)
+		this.set(this.minimum, 'reset')
+	}
+
+	raise() {
+		this.set(this.minimum, 'init')
+	}
+
+	init() {
+		// Ensure we have a gauge init event, can't do in constructor because the parser reference might not be there yet
+		if (this.history.length === 0) {
+			this.history.push({
+				timestamp: this.parser.pull.timestamp,
+				value: this.initialValue,
+				minimum: this.minimum,
+				maximum: this.maximum,
+				delta: 0,
+				reason: 'init',
+			})
+		}
 	}
 
 	/** Modify the current value by the provided amount. Equivalent to `set(currentValue + amount)` */
 	modify(amount: number) {
 		this.set(this._value + amount)
 	}
 
+	/** Increase the current value by the provided amount. */
+	generate = (amount: number) => this.modify(amount)
+
+	/** Decrease the current value by the provided amount. */
+	spend = (amount: number) => this.modify(-amount)
+
 	/** Set the current value of the gauge. Value will automatically be bounded to valid values. Value over the maximum will be tracked as overcap. */
-	set(value: number) {
-		this._value = Math.min(Math.max(value, this.minimum), this.maximum)
+	set(value: number, reason?: GaugeEventReason) {
+		const delta = Math.abs(value - this.value)
+		if (delta === 0 && reason != null) { return }
+
+		if (reason == null) {
+			reason = value > this._value ? 'generate' : 'spend'
+		}
```

Comment:
I kind of agree with the comments ack made in #dev-general when we were first discussing this; it would be better to keep the +/- sign on the delta, so that a consumer can see what's happening with the history without needing to check two fields.  

That may also make the GaugeEventReason field obsolete, if we can treat a reset/init as delta = undefined and then have the backtrack look for those events instead.  But I could see an argument for not turning delta into a tri-state number, whichever works better for the backtrack logic there.

Reply:
I'm going to leave the reason field there, since I'd rather keep delta as a simple numeric, rather than `number?` but yeah, we can make the delta that goes into the history be signed. Still have to Math.abs it for the history correction, but there's no reason that can't be done inside that function.